Test failure:

```
=== RUN   Test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1805e8e]
goroutine 8 [running]:
sync.(*RWMutex).RLock(...)
	/home/travis/.gimme/versions/go1.13.15.linux.amd64/src/sync/rwmutex.go:48
github.com/cilium/cilium/pkg/identity/cache.(*localIdentityCache).lookupByID(0x0, 0xc0000003e8, 0x0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/local.go:170 +0x3e
github.com/cilium/cilium/pkg/identity/cache.(*CachingIdentityAllocator).LookupIdentityByID(0xc0000d70e0, 0x2155cc0, 0xc0000d0018, 0x3e8, 0x10)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/cache.go:248 +0x90
github.com/cilium/cilium/pkg/proxy.(*defaultEndpointInfoRegistry).FillEndpointIdentityByID(0x31a1500, 0x3e8, 0xc0000343d0, 0xc000056b50)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/epinfo.go:50 +0x57
github.com/cilium/cilium/pkg/proxy/logger.(*LogRecord).fillIngressSourceInfo(0xc000034380, 0xc0000343d0, 0xc000698d60, 0x3e8)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/logger/logger.go:161 +0x74
github.com/cilium/cilium/pkg/proxy/logger.logTags.Addressing.func1(0xc000034380)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/logger/logger.go:238 +0x2db
github.com/cilium/cilium/pkg/proxy/logger.(*LogRecord).ApplyTags(0xc000034380, 0xc000698e60, 0x1, 0x1)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/logger/logger.go:290 +0x49
github.com/cilium/cilium/pkg/proxy.(*kafkaRedirect).handleRequest(0xc000362f90, 0xc0002b6cc0, 0xc00026b890, 0xc0002c0080, 0x211ea20, 0xc0002b6c90, 0x3e8, 0xc000056790, 0xf)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:316 +0x3dd
github.com/cilium/cilium/pkg/proxy.(*kafkaRedirect).handleRequests(0xc000362f90, 0xc0000b3800, 0xc0002b6cc0, 0xc0002c0000, 0xc000699e70)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:443 +0x752
github.com/cilium/cilium/pkg/proxy.(*kafkaRedirect).handleRequestConnection(0xc000362f90, 0xc0002b6cc0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:498 +0x1ee
created by github.com/cilium/cilium/pkg/proxy.(*kafkaListener).Listen
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:130 +0x922
FAIL	github.com/cilium/cilium/pkg/proxy	0.049s
FAIL
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.
```

--- a/pkg/datapath/iptables/iptables.go
+++ b/pkg/datapath/iptables/iptables.go
@@ -1178,6 +1178,9 @@ func (m *IptablesManager) addCiliumENIRules() error {
 	nfmask := fmt.Sprintf("%#08x", linux_defaults.MarkMultinodeNodeport)
 	ctmask := fmt.Sprintf("%#08x", linux_defaults.MaskMultinodeNodeport)
 
+	// Note: these rules need the xt_connmark module (iptables usually
+	// loads it when required, unless loading modules after boot has been
+	// disabled).
 	if err := runProg("iptables", append(
 		m.waitArgs,
 		"-t", "mangle",

--- a/pkg/datapath/iptables/iptables.go
+++ b/pkg/datapath/iptables/iptables.go
@@ -1165,6 +1165,10 @@ func (m *IptablesManager) addCiliumNoTrackXfrmRules() error {
 }
 
 func (m *IptablesManager) addCiliumENIRules() error {
+	if !option.Config.EnableIPv4 {
+		return nil
+	}
+
 	nfmask := fmt.Sprintf("%#08x", linux_defaults.MarkMultinodeNodeport)
 	ctmask := fmt.Sprintf("%#08x", linux_defaults.MaskMultinodeNodeport)
 	if err := runProg("iptables", append(

--- a/pkg/datapath/loader/base.go
+++ b/pkg/datapath/loader/base.go
@@ -99,6 +99,43 @@ func writePreFilterHeader(preFilter *prefilter.PreFilter, dir string) error {
 	return fw.Flush()
 }
 
+type setting struct {
+	name      string
+	val       string
+	ignoreErr bool
+}
+
+func addENIRules(sysSettings []setting) ([]setting, error) {
+	// AWS ENI mode requires symmetric routing, see
+	// iptables.addCiliumENIRules().
+	// The default AWS daemonset installs the following rules that are used
+	// for NodePort traffic between nodes:
+	//
+	// # sysctl -w net.ipv4.conf.eth0.rp_filter=2
+	// # iptables -t mangle -A PREROUTING -i eth0 -m comment --comment "AWS, primary ENI" -m addrtype --dst-type LOCAL --limit-iface-in -j CONNMARK --set-xmark 0x80/0x80
+	// # iptables -t mangle -A PREROUTING -i eni+ -m comment --comment "AWS, primary ENI" -j CONNMARK --restore-mark --nfmask 0x80 --ctmask 0x80
+	// # ip rule add fwmark 0x80/0x80 lookup main
+	//
+	// It marks packets coming from another node through eth0, and restores
+	// the mark on the return path to force a lookup into the main routing
+	// table. Without these rules, the "ip rules" set by the cilium-cni
+	// plugin tell the host to lookup into the table related to the VPC for
+	// which the CIDR used by the endpoint has been configured.
+	//
+	// We want to reproduce equivalent rules to ensure correct routing.
+	retSettings := append(sysSettings, setting{"net.ipv4.conf.eth0.rp_filter", "2", false})
+	if err := route.ReplaceRule(route.Rule{
+		Priority: linux_defaults.RulePriorityNodeport,
+		Mark:     linux_defaults.MarkMultinodeNodeport,
+		Mask:     linux_defaults.MaskMultinodeNodeport,
+		Table:    route.MainTable,
+	}); err != nil {
+		return nil, fmt.Errorf("unable to install ip rule for ENI multi-node NodePort: %w", err)
+	}
+
+	return retSettings, nil
+}
+
 // Reinitialize (re-)configures the base datapath configuration including global
 // BPF programs, netfilter rule configuration and reserving routes in IPAM for
 // locally detected prefixes. It may be run upon initial Cilium startup, after
@@ -109,12 +146,6 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner, 
 		mode string
 		ret  error
 	)
-
-	type setting struct {
-		name      string
-		val       string
-		ignoreErr bool
-	}
 
 	args = make([]string, initArgMax)
 
@@ -266,31 +297,9 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner, 
 
 	log.Info("Setting up base BPF datapath")
 
-	// AWS ENI mode requires symmetric routing, see
-	// iptables.addCiliumENIRules().
-	// The default AWS daemonset installs the following rules that are used
-	// for NodePort traffic between nodes:
-	//
-	// # sysctl -w net.ipv4.conf.eth0.rp_filter=2
-	// # iptables -t mangle -A PREROUTING -i eth0 -m comment --comment "AWS, primary ENI" -m addrtype --dst-type LOCAL --limit-iface-in -j CONNMARK --set-xmark 0x80/0x80
-	// # iptables -t mangle -A PREROUTING -i eni+ -m comment --comment "AWS, primary ENI" -j CONNMARK --restore-mark --nfmask 0x80 --ctmask 0x80
-	// # ip rule add fwmark 0x80/0x80 lookup main
-	//
-	// It marks packets coming from another node through eth0, and restores
-	// the mark on the return path to force a lookup into the main routing
-	// table. Without these rules, the "ip rules" set by the cilium-cni
-	// plugin tell the host to lookup into the table related to the VPC for
-	// which the CIDR used by the endpoint has been configured.
-	//
-	// We want to reproduce equivalent rules to ensure correct routing.
 	if option.Config.IPAM == option.IPAMENI {
-		sysSettings = append(sysSettings, setting{"net.ipv4.conf.eth0.rp_filter", "2", false})
-		if err := route.ReplaceRule(route.Rule{
-			Priority: linux_defaults.RulePriorityNodeport,
-			Mark:     linux_defaults.MarkMultinodeNodeport,
-			Mask:     linux_defaults.MaskMultinodeNodeport,
-			Table:    route.MainTable,
-		}); err != nil {
+		var err error
+		if sysSettings, err = addENIRules(sysSettings); err != nil {
 			return fmt.Errorf("unable to install ip rule for ENI multi-node NodePort: %w", err)
 		}
 	}

--- a/pkg/datapath/loader/base.go
+++ b/pkg/datapath/loader/base.go
@@ -123,6 +123,10 @@ func addENIRules(sysSettings []setting) ([]setting, error) {
 	// which the CIDR used by the endpoint has been configured.
 	//
 	// We want to reproduce equivalent rules to ensure correct routing.
+	if !option.Config.EnableIPv4 {
+		return sysSettings, nil
+	}
+
 	retSettings := append(sysSettings, setting{"net.ipv4.conf.eth0.rp_filter", "2", false})
 	if err := route.ReplaceRule(route.Rule{
 		Priority: linux_defaults.RulePriorityNodeport,

